subsys/net/lib/lwm2m/lwm2m_obj_firmware.c

```diff
@@ -298,7 +298,7 @@ static int firmware_update_cb(uint16_t obj_inst_id)
 
 	callback = lwm2m_firmware_get_update_cb();
 	if (callback) {
-		ret = callback(obj_inst_id);
+		ret = callback(obj_inst_id, NULL, 0);
```

Comment:
You could just forward the `payload` and `payload_len` to the application callback.